python/hsfs/storage_connector.py

```diff
@@ -975,17 +967,154 @@ def options(self):
         """Bootstrap servers string."""
         return self._options
 
+    def kafka_options(self):
+        """Return prepared options to be passed to kafka, based on the additional arguments.
+        https://kafka.apache.org/documentation/
+        """
+        config = {}
+
+        # set kafka storage connector options
+        config.update(self.options)
+
+        # set connection properties
+        config.update(
+            {
+                "bootstrap.servers": self.bootstrap_servers,
+                "security.protocol": self.security_protocol,
+            }
+        )
+
+        # set ssl
+        config[
+            "ssl.endpoint.identification.algorithm"
+        ] = self._ssl_endpoint_identification_algorithm
+
+        if self._external_kafka:
+            self._ssl_truststore_location = (
+                client.get_instance()._get_jks_trust_store_path()
+            )
+            self._ssl_truststore_password = client.get_instance()._cert_key
+            self._ssl_keystore_location = (
+                client.get_instance()._get_jks_key_store_path()
+            )
+            self._ssl_keystore_password = client.get_instance()._cert_key
+            self._ssl_key_password = client.get_instance()._cert_key
+
+        if self.ssl_truststore_location is not None:
+            config["ssl.truststore.location"] = self._ssl_truststore_location
+        if self._ssl_truststore_password is not None:
+            config["ssl.truststore.password"] = self._ssl_truststore_password
+        if self.ssl_keystore_location is not None:
+            config["ssl.keystore.location"] = self._ssl_keystore_location
+        if self._ssl_keystore_password is not None:
+            config["ssl.keystore.password"] = self._ssl_keystore_password
+        if self._ssl_key_password is not None:
+            config["ssl.key.password"] = self._ssl_key_password
+
+        return config
+
+    def confluent_options(self):
+        """Return prepared options to be passed to confluent_kafka, based on the provided apache spark configuration.
+        Right now only producer values with Importance >= medium are implemented.
+        https://docs.confluent.io/platform/current/clients/librdkafka/html/md_CONFIGURATION.html
+        """
+        config = {}
+        pem_files_created = False
+        kafka_options = self.kafka_options()
+        for key, value in kafka_options.items():
+            if (
+                key
+                in [
+                    "ssl.truststore.location",
+                    "ssl.truststore.password",
+                    "ssl.keystore.location",
+                    "ssl.keystore.password",
+                ]
+                and not pem_files_created
+            ):
+                (
+                    ca_chain_path,
+                    client_cert_path,
+                    client_key_path,
+                ) = client.get_instance()._write_pem(
```

Comment:
The external client doesn't have this `_write_pem` file, so the call is going to fail.